File: app/src/main/java/com/gxd/design/pattern/behavioral/chain/interceptor/impl/CallServerInterceptor.java
package com.gxd.design.pattern.behavioral.chain.interceptor.impl;

import com.gxd.design.pattern.behavioral.chain.interceptor.Chain;
import com.gxd.design.pattern.behavioral.chain.interceptor.Interceptor;
import com.gxd.design.pattern.behavioral.chain.interceptor.Response;

public class CallServerInterceptor implements Interceptor {
    @Override
    public Response intercept(Chain chain) {
        return new Response() {
        };
    }
}

File: app/src/main/java/com/gxd/design/pattern/behavioral/chain/interceptor/impl/CallServerInterceptor.kt
package com.gxd.design.pattern.behavioral.chain.interceptor.impl

import com.gxd.design.pattern.behavioral.chain.interceptor.Chain
import com.gxd.design.pattern.behavioral.chain.interceptor.Interceptor
import com.gxd.design.pattern.behavioral.chain.interceptor.Response

class CallServerInterceptor : Interceptor {
    override fun intercept(chain: Chain): Response = object : Response {}
}
